internal/app/osd/internal/reg/reg.go

```diff
@@ -271,57 +272,33 @@ func (r *Registrator) Dmesg(ctx context.Context, in *empty.Empty) (data *proto.D
 // be requested and the contents of the log file are streamed in chunks.
 // nolint: gocyclo
 func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err error) {
-	var (
-		client *containerd.Client
-		ctx    context.Context
-		pods   []*pod
-		task   *tasks.GetResponse
-	)
-
-	pods, err = podInfo(req.Namespace)
-	if err != nil {
-		return err
+	var filename string
+
+	switch {
+	case req.Namespace == "system" || req.Id == "kubelet" || req.Id == "kubeadm":
+		filename = filepath.Join("/var/log", req.Id+".log")
```

Comment:
this is a security issue imho: `req.Id` is not sanitized, this way one can read any file (`req.Id = "../../etc/passwd"` just as an example)

Reply:
@bradbeam

Reply:
Yeah I was thinking about that last night. I felt the risk/exposure was minimal given a read-only rootfs and if you have osctl functionality you can own the cluster anyways.
I'll follow up with another PR to clean this up a bit. Would the best approach to be verify against a (static?) list of system services or make use of something like `filepath.Base(req.Id)`?

Reply:
I think `filepath.Base` is the easiest fix to close the hole for now. I'm just afraid that in the future there _might_ be something sensitive we don't want `osctl` to read.

My take on that was a bit more complicated, but still might be implemented: add 'servicelog' API to initd which returns filepath by service id. So that for `logs`, if it's system namespace, we could first try init API to learn filepath by service id. If that fails, fall back to looking through containerd state.

Reply:
^ That is what I think should be the end goal. In the mean time we can use `filepath.Base`. This at least gets us by until the better implementation comes along.